File: dataset/human/python/file_1226.py
def main():
    N = int(input())
    *A, = map(int, input().split())

    def accumulate(a):
        s = 0
        for x in a:
            s += x
            yield s

    *acc, = accumulate(A)
    acc.pop()

    tot = sum(A)

    ans = -1
    for x in acc:
        if ans > (abs_ := abs(tot - x - x)) or not (~ans):
            ans = abs_

    print(ans)


if __name__ == '__main__':
    main()

File: dataset/human/java/file_1226.java
import java.util.*;
public class Main{
 public static void main(String[] args){
   Scanner sc = new Scanner(System.in);
 
   int N = sc.nextInt();
   long[] a =new long[N];
   long[] sumar = new long[N];
   
   long sum =0;
   for(int i=0;i<N;i++) 
   {
     a[i]=sc.nextLong();
     sum += a[i];
     sumar[i]= sum;
   }
   long ans =Math.abs(sum-2*sumar[0]);
   for(int i=1;i<N-1;i++)
   {
     if(Math.abs(sum-2*sumar[i])<ans) ans = Math.abs(sum-2*sumar[i]);
   }
   System.out.println(ans);
 }
}
